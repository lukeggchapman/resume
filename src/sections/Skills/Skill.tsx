import React from 'react'
import { Text, View, StyleSheet } from '@react-pdf/renderer'

import typography from '../../components/typography'

const styles = StyleSheet.create({
  container: {
    marginBottom: 10,
  },
  title: {
    textTransform: 'capitalize',
    marginBottom: 5,
  },
  skill: {
    color: '#333',
  },
})

export interface SkillProps {
  title: string
  skills: string[]
}

const Skill = ({ title, skills }: SkillProps) => (
  <View style={styles.container}>
    <Text style={[typography.h3, styles.title]}>{title}</Text>
    {skills.map(skill => (
      <Text key={skill} style={[typography.h4, styles.skill]}>
        {skill}
      </Text>
    ))}
  </View>
)

export default Skill
