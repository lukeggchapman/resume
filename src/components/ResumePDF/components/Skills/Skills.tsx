import React from 'react'

import { SkillsData } from '../../resumeQuery'
import Skill from './Skill'

export interface SkillsProps {
  data: SkillsData
}

const Skills = ({ data }: SkillsProps) => (
  <>
    {data.edges.map(edge => {
      const skillset = edge.node
      return (Object.keys(skillset) as (keyof typeof skillset)[]).map(
        skillLevel => (
          <Skill
            key={skillLevel}
            title={skillLevel}
            skills={skillset[skillLevel]}
          />
        )
      )
    })}
  </>
)

export default Skills
