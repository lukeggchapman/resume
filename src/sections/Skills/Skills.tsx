import React from 'react'
import Section from 'components/Section'
import skillData from 'content/skills/skills.json'

import Skill from './Skill'

const Skills = () => (
  <Section title="Skills">
    {(Object.keys(skillData) as (keyof typeof skillData)[]).map(
      (skillLevel) => (
        <Skill
          key={skillLevel}
          title={skillLevel}
          skills={skillData[skillLevel]}
        />
      )
    )}
  </Section>
)

export default Skills
