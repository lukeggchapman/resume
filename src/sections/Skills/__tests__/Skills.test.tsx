import React from 'react'
import { shallow } from 'enzyme'

import Skills, { SkillsProps } from '../Skills'

const baseProps: SkillsProps = {
  data: {
    edges: [
      {
        node: {
          beginner: ['skillTest1', 'skillTest2'],
          intermediate: ['skillTest3', 'skillTest4'],
          advanced: ['skillTest5', 'skillTest6'],
        },
      },
    ],
  },
}

describe('Skills', () => {
  const render = (props?: Partial<SkillsProps>) =>
    shallow(<Skills {...{ ...baseProps, ...props }} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
