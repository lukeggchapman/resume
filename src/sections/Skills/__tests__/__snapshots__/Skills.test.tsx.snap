// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<Section
  title="Skills"
>
  <Skill
    key="beginner"
    skills={
      [
        "skillTest1",
        "skillTest2",
      ]
    }
    title="beginner"
  />
  <Skill
    key="intermediate"
    skills={
      [
        "skillTest3",
        "skillTest4",
      ]
    }
    title="intermediate"
  />
  <Skill
    key="advanced"
    skills={
      [
        "skillTest5",
        "skillTest6",
      ]
    }
    title="advanced"
  />
</Section>
`;
